source/frontend/src/features/properties/map/acquisition/common/GenerateForm/hooks/useGenerateAgreement.ts

```diff
@@ -0,0 +1,89 @@
+import { showFile } from 'features/documents/DownloadDocumentButton';
+import { useDocumentGenerationRepository } from 'features/documents/hooks/useDocumentGenerationRepository';
+import { FormTemplateTypes } from 'features/properties/map/shared/content/models';
+import { useApiAcquisitionFile } from 'hooks/pims-api/useApiAcquisitionFile';
+import { useApiContacts } from 'hooks/pims-api/useApiContacts';
+import { AgreementTypes, Api_Agreement } from 'models/api/Agreement';
+import { ExternalResultStatus } from 'models/api/ExternalResult';
+import { Api_GenerateAgreement } from 'models/generate/GenerateAgreement';
+import { Api_GenerateFile } from 'models/generate/GenerateFile';
+
+export const useGenerateAgreement = () => {
+  const { getPersonConcept } = useApiContacts();
+  const { getAcquisitionFile, getAcquisitionFileProperties } = useApiAcquisitionFile();
+  const { generateDocumentDownloadWrappedRequest: generate } = useDocumentGenerationRepository();
+  const generateLetter = async (agreement: Api_Agreement) => {
```

Comment:
nit: having this function retrieve the agreement instead of expecting it to be passed would make it more reusable

Reply:
That is actually a bit of a sticky implementation detail that we should probably think through. with this implementation we generate based on the agreement that is currently visible. If we swapped this to request on click in theory that could result in pulling someone else's concurrent changes. That could either be a good or a bad thing. Might need to talk to the BA's

Reply:
that is fair, however, the other data (what is being retrieved on the hook) does not follow that pattern. As it is it expects the caller to have retrieved only the agreement data previously.